src/Modules/OrchardCore.Commerce/Services/CardPaymentService.cs

```diff
@@ -0,0 +1,157 @@
+using Microsoft.AspNetCore.DataProtection;
+using Microsoft.Extensions.Logging;
+using OrchardCore.Commerce.Abstractions;
+using OrchardCore.Commerce.Extensions;
+using OrchardCore.Commerce.Models;
+using OrchardCore.ContentManagement;
+using OrchardCore.Entities;
+using OrchardCore.Settings;
+using Stripe;
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace OrchardCore.Commerce.Services;
+
+public class CardPaymentService : ICardPaymentService
+{
+    private readonly IShoppingCartPersistence _shoppingCartPersistence;
+    private readonly IPriceService _priceService;
+    private readonly PaymentIntentService _paymentIntentService;
+    private readonly IPriceSelectionStrategy _priceSelectionStrategy;
+    private readonly IContentManager _contentManager;
+    private readonly ISiteService _siteService;
+    private readonly IDataProtectionProvider _dataProtectionProvider;
+    private readonly ILogger<CardPaymentService> _logger;
+
+    public CardPaymentService(
+        IShoppingCartPersistence shoppingCartPersistence,
+        IPriceService priceService,
+        IPriceSelectionStrategy priceSelectionStrategy,
+        IContentManager contentManager,
+        ISiteService siteService,
+        IDataProtectionProvider dataProtectionProvider,
+        ILogger<CardPaymentService> logger)
+    {
+        _paymentIntentService = new PaymentIntentService();
+        _shoppingCartPersistence = shoppingCartPersistence;
+        _priceService = priceService;
+        _priceSelectionStrategy = priceSelectionStrategy;
+        _contentManager = contentManager;
+        _siteService = siteService;
+        _dataProtectionProvider = dataProtectionProvider;
+        _logger = logger;
+    }
+
+    public async Task<PaymentIntent> CreatePaymentAsync(ConfirmPaymentRequest request)
+    {
+        var currentShoppingCart = await _shoppingCartPersistence.RetrieveAsync();
+        var totals = await currentShoppingCart.CalculateTotalsAsync(_priceService, _priceSelectionStrategy);
+
+        // Same here as on the checkout page: Later we have to figure out what to do if there are multiple
+        // totals i.e., multiple currencies.
+        var defaultTotal = totals.FirstOrDefault();
```

Comment:
If totals is empty, is there any point continuing this method execution? Indeed isn't that an error?
Could use an early return or `totals.FirstOrDefault() throw new InvalidOperationException(...)`.
Same for the `defaultTotal` in the other method.